csm/conf/setup.py

```diff
@@ -805,6 +770,9 @@ def init(self, args):
             ha_check = Conf.get(const.CSM_GLOBAL_INDEX, "HA.enabled")
             if ha_check:
                 self._config_cluster(args)
+            cls = self.__class__
+            if cls._is_feature_supported(const.LYVE_PILOT):
+                UDSConfigGenerator.apply()
```

Comment:
This should be done in config phase, not in init.

Reply:
@ajaysrivas All dependencies are being configured in the init phase. Please heck the function calls above this one; those include RabbitMQ, rsyslog, consul, as well FQDN config. This is the reason why I have place UDS config here.

The config phase is dedicated to CSM config itself.

Reply:
However, I agree with Ajay Srivastava that configurations including all dependencies configurations too should be done in config step.
In past we had discussion around it and never came to a common agreement where it should be.
I would suggest lets go ahead with what we have today to maintain the consistency but change it in R1.1 or R2 after discussion.
@ajaysrivas  please suggest

Reply:
Something done wrong earlier can not be an argument to continue doing that.
If there are problems in implementation of configuring UDS & HAproxy in config phase then I am fine with the current implementation but if it's just moving code from init phase to config phase then we should do it. 

Another thing, HAproxy might be starting before CSM init. We should check this with provisioner. If this is the case then we need to restart HAproxy in init phase but the restart will not be required in config phase.

Reply:
I will move these code snippets to the config step if required. Please let me know.

Reply:
@kupranay @ajaysrivas @pawankumarsrivastava I was having a look at the docstrings for `post_install()`, `config()` and `init()`. Here are some important takeaways.

`post_install()` is executed "after just all rpms are install but no service are started"
`config()` performs "configuration for csm" and "is used to move update conf files one time configuration"
`init()` is executed "after all dependency service started"

So you are right this step should not be in `init()`. I am considering moving it to `post_install()` instead of `config()`, though, because it is kind of ensured that it will be executed before dependencies are started. What do you think?

Reply:
@kupranay @ajaysrivas @pawankumarsrivastava I have just realized something. We need access to Elasticsearch to figure out if LyvePilot is a supported feature. As per the documentation above, it would not be available on the post-install phase.

Would it be available on the config phase? If not, we will have to keep UDS config on the init phase, and possibly have to restart UDS and HAproxy explictily.

Reply:
The configuration should be done in config phase.
In config phase, the installation of uds rpm should be checked and it will not require elasticsearch to be started. It could be simple check to find if uds-config.json is present or not. If the config file is present then configure uds and haproxy for uds.

If done in init phase, restart of HAproxy will be required but not the UDS. UDS is started after CSM.

Reply:
I think for now lets not check UDX support. and have it in config stage itself

Reply:
@kupranay @ajaysrivas Requested changes are now complete. I conducted new unit tests and updated the PR original comment with a new test case in order to make sure CSM can contact UDS. Everything is OK.